### PR TITLE
Update Cyphernode, New endpoints, Use better keys, Bearer token script, Express middleware

### PR DESCRIPTION
**Pull request by nk1tz**
### This PR includes a few different things:

- [x] Update to latest cyphernode via replacing `cyphernode/setup.sh` with latest from cyphernode repo.

- [x] New endpoints: `/spend`, `/getNewAddress`. It is now possible to receive and spend funds on the main wallet (aka spending wallet) if your cyphernode's bitcoin core is fully synced.

- [x] new script `get-bearer-token` quality of life improvement to get a bearer token when using things like insomnia.

- [x] Use key=`"003"` for highest levels access to cyphernode api (modifications to `/cyphernode/start.sh` and veto-backend config.

- [x] Use `body-parser` middleware to easily access `req.body` where needed.

- [x] Update veto-frontend typescript to `3.7.4` to match other packages.

- [x] Can receive BTC on dashboard

- [x] Can Send BTC  from dashboard

- [ ] Add to the readme about relevant topics above (would like to discuss these issues with you @drawm ) before deciding what needs documenting :)

### CODE REVIEW

File: packages/veto-backend/src/api/cyphernode/watch.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  address: string
+  eventMessage?: string
+}
+
+export default function watch({ address, eventMessage }: Args): Promise<AxiosResponse> {
+  return cyphernode.post('/watch', {
+    address,
+    // If running cyphernode locally use 'host.docker.internal'
+    // if using the the aws instance of cyphernode, you must use your IP.
+    unconfirmedCallbackURL: 'host.docker.internal:8080/api/notificationcallback',
```
Comment (drawm) [inline packages/veto-backend/src/api/cyphernode/watch.ts]: Since we plan on changing the domain based on environment, add an entry to `config.ts` so we can specify the domain with an environment variable.

File: packages/veto-backend/src/routes/getMempoolInfo.ts [hunk 2]
```diff
@@ -4,12 +4,7 @@ import getMempoolInfo from '../api/cyphernode/getMempoolInfo'
 export default function getMempoolInfoRoute(app: Application): Application {
   app.get('/api/getmempoolinfo', async (request: Request, response: Response) => {
     try {
-      console.log('\n\n\n\n', 'ATTEMPTING GETMEMPOOLINFO', '\n\n\n\n')
```
Comment (drawm) [inline packages/veto-backend/src/routes/getMempoolInfo.ts]: :+1:

File: packages/veto-backend/src/api/cyphernode/watch.ts (hunk 1 above)
Comment (nk1tz) [reply packages/veto-backend/src/api/cyphernode/watch.ts]: good call.  done.

File: cyphernode/.gitignore [hunk 3]
```diff
@@ -14,6 +14,7 @@ config.7z
 config.json
 keys.txt
 cacert.pem
+cyphernode/config.7z-0.2.2
```
Comment (drawm) [inline cyphernode/.gitignore]: leftover?

File: packages/veto-backend/src/api/cyphernode/client.ts [hunk 4]
```diff
@@ -19,8 +19,8 @@ const httpsAgent = new https.Agent({
  * Cyphernode has a weird implementation of JWT at the moment.
  * TODO: Implement standard JWT signing when cyphernode updates this.
  */
-function getBearerToken(key: string, keyId: string): string {
-  const current = Math.round(new Date().getTime() / 1000) + 10
+export function getBearerToken(key: string, keyId: string): string {
```
Comment (drawm) [inline packages/veto-backend/src/api/cyphernode/client.ts]: I don't see it used anywhere else, should it be exported?

File: packages/veto-backend/src/api/cyphernode/client.ts [hunk 5]
```diff
@@ -39,8 +39,8 @@ export default function makeClient(): AxiosInstance {
   if (!client) {
     client = axios.create({
       httpsAgent,
-      baseURL: 'https://localhost:2009/v0/',
-      timeout: 1000,
+      baseURL: config.cyphernode.gatewayUrl,
```
Comment (drawm) [inline packages/veto-backend/src/api/cyphernode/client.ts]: :ok: :fire:

File: packages/veto-frontend/src/pages/Welcome/index.tsx [hunk 6]
```diff
@@ -1,26 +1,19 @@
...
 import { navigate } from '@reach/router'
 import { Flex, Text, Input, Button } from '@veriphi/veto-ui'
 import { mockPassword } from './tempdata.json'
-import axios from 'axios'
 
 export default () => {
   const password = useInput('')
 
-  const handleOnLogin = () => {
+  const handleOnLogin = (e) => {
```
Comment (drawm) [inline packages/veto-frontend/src/pages/Welcome/index.tsx]: Please change to `event` to reduce cognitive burden (need to look at its usage to know its an event)

File: packages/veto-frontend/src/components/molecules/SendBTC/SendBTC.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    setStatus('reviewTx')
+  }
+
+  const handleSend = async () => {
+    setStatus('sending')
+    try {
+      const response = await spend({ address: address.value, amount: amount.value, eventMessage: label.value })
+      setSuccessData({ hash: response.hash, status: response.status })
+      setStatus('successfulSend')
+    } catch (e) {
```
Comment (drawm) [inline packages/veto-frontend/src/components/molecules/SendBTC/SendBTC.tsx]: Its an unpopular opinion but I think all single letter variable name should be avoided (even i,j,k & e).
`i` is a bit of a stretch but can easily be replaced by `index` or `key` or `name` or anything else that represent the value.

`e` can be `error`, `exception`, `event`, `expect` or anything else that starts with an `e`
Without looking at the context/implementation is hard to know what the variable is used for.
Now in the case of `catch(e)` its a bit more obvious because the `catch` tells us its an `error` but it might not be obvious when you read the rest of the catch block.

It could be `setError(event.message)`, `setError(expect.message)`, `setError(environment.message)`, etc.

File: cyphernode/.gitignore (hunk 3 above)
Comment (nk1tz) [reply cyphernode/.gitignore]: I think so, I'll pull this out

File: packages/veto-backend/src/api/cyphernode/client.ts (hunk 4 above)
Comment (nk1tz) [reply packages/veto-backend/src/api/cyphernode/client.ts]: nope my bad

File: packages/veto-frontend/src/components/molecules/SendBTC/SendBTC.tsx (hunk 7 above)
Comment (nk1tz) [reply packages/veto-frontend/src/components/molecules/SendBTC/SendBTC.tsx]: Great case! I will adopt the standard :)

File: packages/veto-backend/src/utils/config.ts [hunk 8]
```diff
@@ -48,17 +48,19 @@ export default {
   ...baseConfig,
   cyphernode: {
     // Gateway url used to communicate with cyphernode's ecosystem
-    gatewayUrl: 'https://localhost:2009/v0/',
+    // gatewayUrl: 'https://localhost:2009/v0/',
```
Comment (drawm) [inline packages/veto-backend/src/utils/config.ts]: Leftover?

File: packages/veto-backend/src/utils/config.ts [hunk 9]
```diff
@@ -48,17 +48,19 @@ export default {
...
+    gatewayUrl: process.env.CYPHERNODE_URL ?? 'https://ec2-54-145-103-133.compute-1.amazonaws.com:2009/v0/',
```
Comment (drawm) [inline packages/veto-backend/src/utils/config.ts]: Do you know if it's a static domain/ip? If not it will change at some point in the future.

No meed to change it roght away but its something to keep in mind in the future

File: packages/veto-frontend/src/types/axios.d.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import axios from 'axios'
+
```
Comment (drawm) [inline packages/veto-frontend/src/types/axios.d.ts]: Why is this needed? 
Axios already has type definition https://github.com/axios/axios/blob/master/index.d.ts

File: packages/veto-frontend/src/api/index.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import axios from 'axios'
+
+export const getNewAddress = async (): Promise<string> => {
+  const { data }: { data: { address: string } } = await axios(`http://${window.location.host}/api/getnewaddress`)
```
Comment (drawm) [inline packages/veto-frontend/src/api/index.ts]: I think you can do ` const { data } = await axios.get<Promise<{data: { address: string }}>>('http://${window.location.host}/api/getnewaddress')` and you wont need the extra type definition for axios

File: packages/veto-backend/src/utils/config.ts (hunk 9 above)
Comment (nk1tz) [reply packages/veto-backend/src/utils/config.ts]: Will keep in mind. I believe static though.
